test/functional/wallet_backwards_compatibility.py

```diff
@@ -161,6 +162,38 @@ def test_v22_inactivehdchain_path(self):
         except ImportError:
             self.log.warning("sqlite3 module not available, skipping lack of keymeta records check")
 
+    def test_ignore_legacy_during_startup(self, legacy_nodes, node_master):
+        self.log.info("Test that legacy wallets are ignored during startup on v29+")
+
+        legacy_node = legacy_nodes[0]
+        wallet_name = f"legacy_up_{legacy_node.version}"
+        legacy_node.loadwallet(wallet_name)
+        legacy_wallet = legacy_node.get_wallet_rpc(wallet_name)
+
+        # Move legacy wallet to latest node
+        wallet_path = node_master.wallets_path / wallet_name
+        wallet_path.mkdir()
+        legacy_wallet.backupwallet(wallet_path / "wallet.dat")
+        legacy_wallet.unloadwallet()
+
+        # Write wallet so it is automatically loaded during init
+        settings_path = node_master.chain_path / "settings.json"
+        with settings_path.open("w") as fp:
+            json.dump({"wallet": [wallet_name]}, fp)
+
```

Comment:
In 86e1111239cdb39dd32cfb5178653c608fa30515 "test: verify node skips loading legacy wallets during startup"

Is there a reason to use settings.json instead of passing in `-wallet=legacy_up...` as part of `extra_args` during the restart?

Reply:
> Is there a reason to use settings.json instead of passing in `-wallet=legacy_up...` as part of `extra_args` during the restart?

I just wanted to make the test case as realistic as possible (an existing legacy wallet with `load_on_startup=true`). But there shouldn't be any difference.